src/lava/networks/gradedvecnetwork.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (C) 2022-23 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+from scipy.sparse import csr_matrix
+
+from lava.proc.graded.process import InvSqrt
+from lava.proc.graded.process import NormVecDelay
+from lava.proc.sparse.process import Sparse
+from lava.proc.dense.process import Dense
+from lava.proc.prodneuron.process import ProdNeuron
+from lava.proc.graded.process import GradedVec
+from lava.proc.lif.process import LIF
+from lava.proc.io import sink, source
+from lava.proc import embedded_io as eio
+
+from lava.magma.core.process.variable import Var
+from lava.magma.core.process.ports.ports import InPort, OutPort
+
+from .network import AlgebraicVector, AlgebraicMatrix
+
+
+class InputVec(AlgebraicVector):
+    """InputVec
+    Simple input vector. Adds algebraic syntax to RingBuffer
+
+    Parameters
+    ----------
+    vec: np.ndarray
+        NxM array of input values. Input will repeat every M steps.
+    loihi2=False: bool
```

Comment:
No need to specify the `=False` here. This should be visible in the signature of `__init__`.

Also, if the `loihi2` parameter is optional (i.e has a default value as in this case), it's docstring should look like:

`loihi: bool, optional`